src/calendar.service.ts

```diff
@@ -159,4 +161,45 @@ export class CalendarService {
     update() {
         this.slideUpdated.next();
     }
+
+    getEventOccurences(event: IEvent, utcStartTime: number, utcEndTime: number): IOccurence[] {
+        let occurences: IOccurence[] = []
+        if( event.startTime > event.endTime) {
+            return occurences
+        }
+        if (!event.rruleFreq) {
+            let eventUTCStartTime: number
+            let eventUTCEndTime: number
+            if (event.allDay) {
+                eventUTCStartTime = event.startTime.getTime()
+                eventUTCEndTime = event.endTime.getTime()
+            } else {
+                    eventUTCStartTime = Date.UTC(event.startTime.getFullYear(), event.startTime.getMonth(), event.startTime.getDate())
+                    eventUTCEndTime = Date.UTC(event.endTime.getFullYear(), event.endTime.getMonth(), event.endTime.getDate() + 1)
+            }
+            if( eventUTCEndTime > utcStartTime && eventUTCStartTime < utcEndTime ) {
+                occurences.push({
+                    eventUTCStartTime: eventUTCStartTime,
+                    eventUTCEndTime: eventUTCEndTime
+                })
+            }
+            return occurences
+        }
+        if(event.rruleUntil && event.rruleUntil.getTime() < utcStartTime ){
+            return occurences;
+        }
+        let time = utcStartTime + 1
+        let oneDay = 86400000
+        let recurrence = moment(event.startTime.getDate()).recur().every(event.rruleInterval, event.rruleFreq)
+        while (time < utcEndTime + 1) {
```

Comment:
I think it's better to use the moment-rs built-in function to generate the dates.

> 
With both a start date and an end date set, you can generate all dates within that range that match the pattern (including the start/end dates).

let recurrence = moment()
  .recur('01/01/2014', '01/07/2014')
  .every(2)
  .days()

// Outputs: ["01/01/2014", "01/03/2014", "01/05/2014", "01/07/2014"]
allDates = recurrence.all('L')